### ISSUE TITLE
Upgrade com.sun.tdk.signaturetest.SignatureTest in sigtest-maven-plugin

### ISSUE THREAD

**Issue by alwin-joseph** [eclipse-ee4j/jakartaee-tck-tools#29]
Upgrade com.sun.tdk.signaturetest.SignatureTest in sigtest-maven-plugin to be executed as a maven goal.

**Context:**
The signature tests in platform-tck and several standalone TCKs that were separated out, uses a legacy signaturetest framework majorly to invoke the sigtest-maven-plugin with specific arguments.
For eg: The [runSignatureTest method](https://github.com/jakartaee/platform-tck/blob/d5e5665fa57579379e5b7b43e823581041df6aa2/signaturetest/src/main/java/com/sun/ts/tests/signaturetest/SigTestDriver.java#L163C1-L195C74) is used to invoke the com.sun.tdk.signaturetest.SignatureTest with the right arguments to run the Signature Tests.

The [org/netbeans/apitest/SigtestCheck.java](https://github.com/eclipse-ee4j/jakartaee-tck-tools/blob/master/tools/sigtest/src/main/java/org/netbeans/apitest/SigtestCheck.java) can be run using maven goal "check" specified like [here](https://github.com/jakartaee/platform-tck/blob/d5e5665fa57579379e5b7b43e823581041df6aa2/glassfish-runner/cdi-tck/pom.xml#L464C1-L483C22) for CDI TCK.

**Expected:**
Can we also improve the com.sun.tdk.signaturetest.SignatureTest so as to execute the same via maven goal and parameters specified. This can avoid the complexities of the legacy signaturetest module in platform-tck and dependency on it.

cc @scottmarlow @gurunrao @arjantijms @jamezp

**Comment by scottmarlow** [comment]
Is this needed for the Expression Language 6.0 TCK?

Note that [SigtestHandler](https://github.com/eclipse-ee4j/jakartaee-tck-tools/blob/master/tools/sigtest/src/main/java/org/netbeans/apitest/SigtestHandler.java) is run from https://github.com/eclipse-ee4j/jakartaee-tck-tools/blob/master/tools/sigtest/src/main/java/org/netbeans/apitest/SigtestCheck.java.

The ignored jdk classes list should be checked in [src/main/java/org/netbeans/apitest/SigtestHandler.java](https://github.com/eclipse-ee4j/jakartaee-tck-tools/blob/master/tools/sigtest/src/main/java/org/netbeans/apitest/SigtestHandler.java#L99) which should cover the equivalent of -IgnoreJDKClass

There is some support already for capturing the log output via [org.netbeans.apitest.SigtestHandler](https://github.com/eclipse-ee4j/jakartaee-tck-tools/blob/master/tools/sigtest/src/main/java/org/netbeans/apitest/SigtestHandler.java#L146) and below code but could use more changes.

For reference the general documentation is here https://github.com/eclipse-ee4j/jakartaee-tck-tools/tree/master/tools/sigtest.

**Comment by jamezp** [comment]
TBH I think we need to think about how these are ran in general. With the maven plugin the class path is setup with [Maven dependencies from the class path](https://github.com/eclipse-ee4j/jakartaee-tck-tools/blob/5dd621604fe6a12bc2c53e4df898194f866ea903/tools/sigtest/src/main/java/org/netbeans/apitest/SigtestCheck.java#L223-L227). IMO that's wrong as it should only be using dependencies defined in an explicit class path. In other words, dependencies from a server not Maven. We need to verify the signature tests are ran against the libraries the server is shipped with.

That said, when I was doing some testing in the jakarta/rest TCK, the signature tests always passed for me even though they failed with the old manual test method. I didn't dig too much into it, but before we do that we should definitely look into why the check would pass.

**Comment by scottmarlow** [comment]
I agree with @jamezp that the Jakarta EE Platform TCK signature tests need to deploy to all of the same EE containers that we test with currently in earlier EE releases.  

So for Common Annotations I think that only needs a Java SE environment (e.g. no Platform or Web Profile implementation).  

The same for the Standalone Expression Language TCK but the Expression Language tests (e.g. not the signature test) will definitely need to deploy to Web containers for the Platform/Web Profile TCK test runs.

For the Platform/Web Profile TCK signature test runs, we definitely need to deploy the signature tests to EE containers as we have done in the past.

**Comment by scottmarlow** [comment]
> > Is this needed for the Expression Language 6.0 TCK?
> 
> This enhancement is not a mandatory requirement. This was motivated from the below [comment](https://eclipsefoundationhq.slack.com/archives/C0131MLD538/p1707924410037959).

From https://wildfly.zulipchat.com/#narrow/stream/242838-Jakarta-EE/topic/Signature.20TCK.20testing thread, James mentioned the problems with the maven sigtest plugin.  

For `static` testing (e.g. Java SE style classpath specified) we talked about adding a Maven Plugin property for passing the (static) classpath property in (we need to a unique name for that property that is not likely to be used for other things, classpath seems too generic).

For `reflective` testing (e.g. deploying signature tests to an EE container, I don't think the Maven Plugin will control the deployment.   

> 
> _Also, in Faces and REST there is an enormous amount of extra code for checking the signature. Is that really needed? Why doesn’t the maven plugin do the check?_
> 

+1

**Comment by scottmarlow** [comment]
fyi https://github.com/eclipse-ee4j/jakartaee-tck-tools/issues/158 is for updating the Generate Maven plugin to have a way to exclude a subpackage in one of the specified packages.
